lib/elixir_search_extractor/file_upload.ex

```diff
@@ -0,0 +1,90 @@
+defmodule ElixirSearchExtractor.FileUpload do
+  import Ecto.Query, warn: false
+
+  alias ElixirSearchExtractor.FileUpload.KeywordFile
+  alias ElixirSearchExtractor.Repo
+
+  def list_keyword_files do
+    Repo.all(KeywordFile)
+  end
+
+  def create_keyword_file(attributes, user_id) do
+    csv_file = attributes["csv_file"]
+
+    case validate_file(csv_file) do
+      {:ok} ->
+        attributes = Map.put(attributes, "csv_file", upload_file_path(user_id, csv_file))
+        attributes = Map.put(attributes, "user_id", user_id)
+
+        %KeywordFile{}
+        |> KeywordFile.changeset(attributes)
+        |> Repo.insert()
+        |> upload_file(user_id, csv_file)
```

Comment:
How about uploading the file before creating the record because it might be possible that the record is created while the upload is failed 🤔  Then the job to process the file cannot find the file or we will handle that case in the background job?

 I can imagine having something like this 💭 

```elixir
with :ok <- validate_file(csv_file),
     {:ok, upload_file_path} <- upload_file(user_id, csv_file) ->
  attributes = 
    attributes
    |> Map.put("csv_file", upload_file_path)
    |> Map.put("user_id", user_id)

  %KeywordFile{}
  |> KeywordFile.changeset(attributes)
  |> Repo.insert()
else
  {:error, reason} -> {:error, reason}
end 
```

With this later, we can refactor validate and upload functions out to new modules. Just an idea, feel free to ignore this 👍